src/utils/rfqm_db_utils.ts

```diff
@@ -141,6 +142,32 @@ export class RfqmDbUtils {
         return entity;
     }
 
+    public async upsertRfqmWorkerHeartbeatToDbAsync(
+        address: string,
+        index: number,
+        balance: BigNumber,
+    ): Promise<RfqmWorkerHeartbeatEntity> {
+        if (!Number.isInteger(index)) {
+            throw new Error(`Index ${index} is not an integer`);
+        }
+        const repository = this._connection.getRepository(RfqmWorkerHeartbeatEntity);
+
+        // Why I did not use `.save`:
+        // The `rfqm_worker_heartbeat` table has a trigger to automatically update the timestamp on UPDATE
+        // but the `.save` functionality is smart enough to not actually execute the update if none of the
+        // data has changed. Since this only happens when a worker balance changes, the timestamp won't
+        // update unless `.update` is explicitly called.
+        const updatedEntity = await repository.preload({ address, index, balance });
+        if (updatedEntity !== undefined) {
+            await this._connection.getRepository(RfqmWorkerHeartbeatEntity).update(address, updatedEntity);
+            return updatedEntity;
+        }
+
+        const newEntity = new RfqmWorkerHeartbeatEntity({ address, index, balance });
+        await this._connection.getRepository(RfqmWorkerHeartbeatEntity).insert(newEntity);
```

Comment:
Does this perform an upsert? also, how are you saving `balance`? `balance` is a BigInt, which usually is persisted to disk as a string. This means that any query won't be doing numerical comparison but rather lexicographical comparison.

Have we considered saving this number as `decimal`, and simply converting BigInt to decimal format?